ironic/drivers/modules/drac/management.py

```diff
@@ -357,3 +358,32 @@ def get_sensors_data(self, task):
                   sensor type, which can be processed by Ceilometer.
         """
         raise NotImplementedError()
+
+    @METRICS.timer('DracManagement.reset_idrac')
+    @base.clean_step(priority=CONF.drac.clean_priority_reset_idrac)
+    def reset_idrac(self, task):
+        """Reset the iDRAC configuration.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        node = task.node
+
+        client = drac_common.get_drac_client(node)
+        client.reset_idrac(force=True, wait=True)
+
+    @METRICS.timer('DracManagement.known_good_state')
+    @base.clean_step(priority=CONF.drac.clean_priority_known_good_state)
+    def known_good_state(self, task):
+        """Reset the iDRAC, Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        node = task.node
+
+        client = drac_common.get_drac_client(node)
+        client.reset_idrac(force=True, wait=True)
+        client.delete_jobs(job_ids=['JID_CLEARALL_FORCE'])
```

Comment:
I believe you need to tell ironic conductor to resume cleaning after the reset completes like this:
manager_utils.notify_conductor_resume_clean(task)

Reply:
I have added line :  agent_base_vendor._notify_conductor_reume_clean(task)  after delete jobs completes, but while executing cleaning steps, "Failed to tear down from cleaning" exception occurs at this line  -https://github.com/cdearborn/ironic/blob/master/ironic/conductor/manager.py#L1443 and node got stuck into CLEANING state.

Reply:
I would try calling resume_cleaning the same way that the RAID creation cleaning step does here:
https://github.com/cdearborn/ironic/blob/master/ironic/drivers/modules/drac/raid.py#L903
I'm not sure if the call you are making is doing the same things or not.